File: src/02_Control_Flow/Conditionals_Switch.cpp
/**
 * Level 2: Control Flow & Logic
 * File: Conditionals_Switch.cpp
 * Topic: If-else branching, ternary operators, and switch statements
 */

#include <iostream>

int main() {
    int score = 85;

    // Ternary Operator: (condition) ? if_true : if_false
    std::string result = (score >= 50) ? "PASS" : "FAIL";
    std::cout << "Exam Result: " << result << "\n";

    // Switch case statement
    char grade = 'B';
    switch (grade) {
        case 'A': std::cout << "Excellent!\n"; break;
        case 'B': std::cout << "Good Job!\n"; break;
        case 'C': std::cout << "Average\n"; break;
        default:  std::cout << "Needs Improvement\n"; break;
    }

    return 0;
}

File: src/02_Control_Flow/Conditionals_Switch.java
/**
 * Level 2: Control Flow & Logic
 * File: Conditionals_Switch.java
 * Topic: If-else branching, ternary operators, and modern switch expressions
 */

public class Conditionals_Switch {
    public static void main(String[] args) {
        int score = 85;

        // Ternary Operator
        String result = (score >= 50) ? "PASS" : "FAIL";
        System.out.println("Exam Result: " + result);

        // Modern Switch (Java 14+)
        char grade = 'B';
        switch (grade) {
            case 'A' -> System.out.println("Excellent!");
            case 'B' -> System.out.println("Good Job!");
            case 'C' -> System.out.println("Average");
            default  -> System.out.println("Needs Improvement");
        }
    }
}
